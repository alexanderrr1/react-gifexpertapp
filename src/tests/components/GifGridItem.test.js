import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe('Pruebas para <GifGridItem />', () => {

    const title = 'title123';
    const url = 'https://prueba.com';
    
    const wrapper = shallow( 
        <GifGridItem 
            title={ title }
            url={ url } 
        />
    );

    test('Debe de mostrar el componente GifGridItem de manera correcta', () => {  

        expect( wrapper ).toMatchSnapshot();   

    });

    test('Debe de tener un pÃ¡rrafo con el title', () => {
        
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );

    });

    test('Debe de tener la imagen igual al url y alt de los props', () => {
        
        const img = wrapper.find('img');
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );

    });

    test('Debe de tener animate__fadeIn', () => {
        
        const div = wrapper.find('div');
        expect( div.prop('className') ).toContain('animate__fadeIn');

    })
    
    
    
    
})
