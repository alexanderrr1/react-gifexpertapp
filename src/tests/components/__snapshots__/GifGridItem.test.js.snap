// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para <GifGridItem /> Debe de mostrar el componente GifGridItem de manera correcta 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="title123"
    src="https://prueba.com"
  />
  <p>
     
    title123
     
  </p>
</div>
`;
